### PR TITLE
[SG-32130] Wildcard: `Collapse` migration

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Description
Migrate `reactstrap Collapse` to `wildcard Collapse`

## Implementation
Migrate all the `reactstrap Collapse` implementation from the codebase to use `wildcard Collapse`

## Refs
- [Sourcegraph Issue](https://github.com/sourcegraph/sourcegraph/issues/32130)
- [Gitstart Ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-32130)

## Test plan
Compare the UI of changed component of this branch with the main branch

## App preview:
- [Link](https://sg-web-contractors-sg-32130-25rv.onrender.com)

### CODE REVIEW

**Review by oleggromov — APPROVED** [review]
Looks good except one unused var. Please have a look at the comment

File: client/search-ui/src/results/sidebar/SearchSidebarSection.tsx [hunk 1]
```diff
@@ -91,41 +90,50 @@ export const SearchSidebarSection: React.FunctionComponent<{
         }
 
         const [collapsed, setCollapsed] = useState(startCollapsed)
+        const handleOpenChange = useCallback(
+            collapsed => {
```
[the remark this answers is not in the document]
Comment (gitstart-sourcegraph) [reply client/search-ui/src/results/sidebar/SearchSidebarSection.tsx]: @oleggromov Thanks for reviewing, we applied changes for this

File: client/search-ui/src/results/sidebar/SearchSidebar.tsx [hunk 2]
```diff
@@ -110,7 +110,7 @@ export const SearchSidebar: React.FunctionComponent<SearchSidebarProps> = props
             setCollapsedSections(openSections => {
                 const newSettings: TemporarySettings['search.collapsedSidebarSections'] = {
                     ...openSections,
-                    [id]: open,
+                    [id]: !open,
```
Comment (gitstart-sourcegraph) [inline client/search-ui/src/results/sidebar/SearchSidebar.tsx:113]: We are saving setting `collapsed` states in `search.collapsedSidebarSections` => using `open` doesn't make sense, so I updated it to use `!open`

File: client/search-ui/src/results/sidebar/SearchSidebarSection.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (gitstart-sourcegraph) [reply client/search-ui/src/results/sidebar/SearchSidebarSection.tsx]: @oleggromov Thanks for reviewing, added some changes to resolve this.

**Comment by gitstart-sourcegraph** [comment]
@oleggromov This PR is ready for review again :)

File: client/search-ui/src/results/sidebar/SearchSidebar.tsx (hunk 2 above)
Comment (oleggromov) [reply client/search-ui/src/results/sidebar/SearchSidebar.tsx:113]: It occurs to me that this started working incorrectly after another inversion in `SearchSidebarSection` or perhaps a bunch of other changes from `collapse` to `open` in many components. 

Anyways, the only thing that is a bit bothering is that after this update, users will get the opposite of their settings. But I guess this isn't a big deal - not worse than adding code that inverses everything once after a release.

**Review by oleggromov — APPROVED** [review]

**Pull request merged**